src/complex/DataStructure/AbstractDataStore.hpp

```
@@ -506,12 +509,145 @@ class AbstractDataStore : public IDataStore
+   */
+  void setTuple(index_type tupleIndex, const std::vector<value_type>& values)
+  {
+    if(values.size() != getNumberOfComponents())
+    {
+      return;
+    }
+
+    if(tupleIndex > getNumberOfTuples())
+    {
+      return;
+    }
+
+    index_type numComponents = getNumberOfComponents();
+    index_type offset = tupleIndex * numComponents;
+    for(index_type i = 0; i < numComponents; i++)
+    {
+      setValue(offset + i, values[i]);
+    }
+  }
+
+  /**
+   * @brief Sets all component values for a tuple using a span of values.
+   *
+   * If the tuple index is out of bounds or the provided span does not match
+   * the number of components, this method does nothing.
+   * @param tupleIndex
+   * @param values
+   */
+  void setTuple(index_type tupleIndex, const nonstd::span<value_type>& values)
+  {
+    if(values.size() != getNumberOfComponents())
+    {
+      return;
+    }
+
+    if(tupleIndex > getNumberOfTuples())
+    {
+      return;
+    }
+
+    index_type numComponents = getNumberOfComponents();
+    index_type offset = tupleIndex * numComponents;
+    for(uint64 i = 0; i < numComponents; i++)
+    {
+      setValue(offset + i, values[i]);
+    }
+  }
+
+  /**
+   * @brief Sets the component value using a given tuple and component index.
+   *
+   * This method does nothing if the tuple or component indices are out of bounds
+   * @param tupleIndex
+   * @param componentIndex
+   * @param value
+   */
+  void setComponent(index_type tupleIndex, index_type componentIndex, value_type value)
+  {
+    if(tupleIndex > getNumberOfTuples())
+    {
+      return;
+    }
+
+    if(componentIndex > getNumberOfComponents())
+    {
+      return;
+    }
+
+    index_type index = tupleIndex * getNumberOfComponents() + componentIndex;
+    setValue(index, value);
+  }
+
+  /**
+   * @brief Returns the component value at the specified tuple and component index.
+   *
+   * This method returns the default T value if either index is out of bounds.
```

I think we should fail somehow instead of returning a default value.